packages/docs/community/release-process/smart-contracts.md

```diff
@@ -34,26 +29,53 @@ Ongoing smart contract development is done on the `master` branch.
+
+## Script 2 ##
+# generate report on backwards compatibility between RC1 and current branch with new release contracts
+yarn ts-node scripts/check-backward.ts report --exclude ".*Test|Mock.*|I[A-Z].*|.*Proxy|LinkedList|SortedLinkedList|SortedLinkedListWithMedian|MultiSig.*|ReleaseGold" -o build/rc1 -n build/contracts --output_file report.json
 
-This script does the following:
+## Script 3 ##
+# deploy new contracts and generate governance proposal for upgrade
+yarn truffle exec --network development ./scripts/truffle/make-release.js --build_directory build/ --report report.json --proposal proposal.json --initialize_data example-initialize-data.json'      
+```
 
-  1. Compiles the contracts at `$CURRENTLY_RELEASED_BRANCH` and confirms that the compiled bytecode matches what is currently deployed on the specified network.
-  2. Compiles the contracts in the current branch and checks backwards compatibility with what is currently deployed on the specified network, with the following exceptions:
+These scripts do the following:
+  #### Script 1:
+  1. Compiles the contracts at <code>rc1</code> branch and confirms that the compiled bytecode matches what is currently deployed on the specified network.
+  #### Script 2:
+  1. Compiles the contracts in the current branch and checks backwards compatibility with what is currently deployed on the specified network, with the following exceptions:
      1. If the STORAGE version has changed, does not perform backwards compatibility checks
      2. If the MAJOR version has changed, checks that the storage layout is backwards compatible, but does not check that the contract ABI is backwards compatible.
-  3. For contracts that have changed, confirms that the version number in the current branch is strictly greater than the deployed version number.
-  4. For contracts that have not changed, confirms that the version number in the current branch is exactly the same as the deployed version number
-  5. For contracts that have changed, deploys those contracts to the specified network.
-  6. Creates and submits a single governance proposal to upgrade to the newly deployed contracts.
+  2. For contracts that have changed, confirms that the version number in the current branch is strictly greater than the deployed version number.
+  3. For contracts that have not changed, confirms that the version number in the current branch is exactly the same as the deployed version number
+  
+  #### Script 3:
+  1. For contracts that have changed, deploys those contracts to the specified network.
+  2. Creates and submits a single governance proposal to upgrade to the newly deployed contracts.
      1. STORAGE updates are adopted by deploying a new proxy and implementation and updating the Registry contract.
      2. All other updates are adopted by updating the proxy contract’s implementation pointer.
 
+## Verify Release Process
```

Comment:
What do you think about referencing the audit commits here? I.e. if I understand correctly, we should be asserting that the target network has the byte code of audited commit `i-1` and that the proposal yields audited commit `i`?

Reply:
In general for this section with the scripts, it might be nice to give some higher-level context. I'd imagine not everyone will be running these scripts, but even for those folks, it would be nice what the dangers/risks these scripts are trying to protect against (primarily to get explicit compatibility checks and an audit trail)

Reply:
in reference to commits: I'm a tiny bit confused by what exactly you're proposing. On our end, there's the audited commit `i` and then as much as 20 cherry-picked commits with audit fixes (or other random stuff like makings sure the build isn't failing bc of some new build spec). I figure by the time we make the proposal there's branch `release/contract/N` at commit `abc123` that is our official proposal. and the `rci` branch with commit `def456` with the existing contracts

Reply:
So in this case `def456` is the audited commit `i-1`, i.e. the commit from the last audit. Users should verify that the current byte code reflects `i-1` (i.e. `rc1` right now) right?